### PR TITLE
[SUBS-1455] Add validation support for BAM and CRAM files

### PR DESCRIPTION
**Pull request by ke4**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java [hunk 1]
```diff
@@ -1,24 +1,24 @@
 package uk.ac.ebi.subs.filecontentvalidator.service;
 
-import com.rabbitmq.client.Command;
 import lombok.Getter;
 import lombok.NonNull;
 import lombok.RequiredArgsConstructor;
 import org.springframework.amqp.rabbit.core.RabbitMessagingTemplate;
 import org.springframework.stereotype.Service;
 import uk.ac.ebi.subs.filecontentvalidator.config.CommandLineParams;
+import uk.ac.ebi.subs.filecontentvalidator.validators.FastqFileValidator;
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java]: Do we need these validator imports?

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/validators/BamCramFileValidator.java [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    void executeValidation(Path outputDirectory) throws IOException, InterruptedException {
+        String command = validatorCommandLine(outputDirectory);
+        LOGGER.info("command: {}", command);
+        Process process = Runtime.getRuntime().exec(command);
+        process.waitFor();
+    }
+
+    String validatorCommandLine(Path outputDirectory) {
```
Comment (davidrichardson) [inline src/main/java/uk/ac/ebi/subs/filecontentvalidator/validators/BamCramFileValidator.java]: Should we just use quick check, or use quick check followed by `view -c` ?

File: src/test/java/uk/ac/ebi/subs/filecontentvalidator/validators/VcfFileValidatorTest.java [hunk 3]
```diff
@@ -74,7 +76,7 @@ public void test_file_finding() throws IOException {
 
     @Test
     public void test_outputDir_is_a_directory() throws IOException {
```
Comment (davidrichardson) [inline src/test/java/uk/ac/ebi/subs/filecontentvalidator/validators/VcfFileValidatorTest.java]: should this test move out of VcfFileValidatorTest, as it's now testing a method in a different class?

**Review by davidrichardson — CHANGES_REQUESTED** [review]
This looks good, I just have some minor comments 👍

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java (hunk 1 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/service/FileContentValidationHandler.java]: You are right, I will remove them

File: src/main/java/uk/ac/ebi/subs/filecontentvalidator/validators/BamCramFileValidator.java (hunk 2 above)
Comment (ke4) [reply src/main/java/uk/ac/ebi/subs/filecontentvalidator/validators/BamCramFileValidator.java]: OK, I will implement it

File: src/test/java/uk/ac/ebi/subs/filecontentvalidator/validators/VcfFileValidatorTest.java (hunk 3 above)
Comment (ke4) [reply src/test/java/uk/ac/ebi/subs/filecontentvalidator/validators/VcfFileValidatorTest.java]: I agree. I can move this and test_file_finding method to a separate test class

**Review by davidrichardson — APPROVED** [review]

**Pull request merged**